javascript/react/wagmi.sh/src/DeployLock.js

```diff
@@ -0,0 +1,143 @@
+import * as React from 'react'
+import { ethers } from 'ethers'
+import { UnlockV11, PublicLockV11 } from "@unlock-protocol/contracts";
+
+import {
+  erc20ABI,
+  useAccount,
+  useSendTransaction,
+  useWaitForTransaction,
+  useContractRead,
+  usePrepareContractWrite
+} from 'wagmi'
+
+const lockInterface = new ethers.utils.Interface(PublicLockV11.abi)
+
+
+export function DeployLock() {
+  const { address: creator } = useAccount()
+
+  const [calldata, setCalldata] = React.useState('')
+  const [name, setName] = React.useState('My Membership')
+  const [price, setPrice] = React.useState(1)
+  const [duration, setDuration] = React.useState(30) // in days
+  const [supply, setSupply] = React.useState(10000)
+  const [currency, setCurrency] = React.useState('') // address of the ERC20. If 0x0, uses base currency
+
+  const { data: decimals } = useContractRead({
+    addressOrName: currency,
+    contractInterface: erc20ABI,
+    functionName: 'decimals',
+    enabled: currency !== ethers.constants.AddressZero
+  })
+
+
+  React.useEffect(() => {
+    const prepareCalldata = async () => {
+      setCalldata(lockInterface.encodeFunctionData(
+        'initialize(address,uint256,address,uint256,uint256,string)',
+        [
+          creator,
+          duration * 60 * 60 * 24, // duration is in days!
+          currency || ethers.constants.AddressZero,
+          ethers.utils.parseUnits(price.toString(), decimals || 18),
+          supply,
+          name,
+        ]
+      ))
+    }
+    prepareCalldata()
+  }, [creator, duration, currency, price, supply, name, decimals])
+
+
+  const { config } = usePrepareContractWrite({
+    addressOrName: '0x627118a4fB747016911e5cDA82e2E77C531e8206',
+    contractInterface: UnlockV11.abi,
+    functionName: 'createUpgradeableLockAtVersion',
+    args: [calldata, 11] // We currently deploy version 11
+  })
```

Comment:
Since this is an example, I think we should add more comment on what this is doing and explain.